native/src/main/java/io/ballerina/stdlib/http/api/HttpService.java

```diff
@@ -258,9 +262,108 @@ private static void processResources(HttpService httpService) {
         } else {
             log.debug("OpenAPI definition is not available");
         }
+        processLinks(httpService, httpResources);
         httpService.setResources(httpResources);
     }
 
+    private static void processLinks(HttpService httpService, List<HttpResource> httpResources) {
+        for (HttpResource targetResource : httpResources) {
+            for (HttpResource.LinkedResource link : targetResource.getLinkedResources()) {
+                HttpResource linkedResource = null;
+                for (HttpResource resource : httpResources) {
+                    linkedResource = getLinkedResource(link, linkedResource, resource);
+                }
+                if (linkedResource != null) {
+                    setLinkToResource(httpService, targetResource, link, linkedResource);
+                } else {
+                    String msg = "cannot find" + (link.getMethod() != null ? " " + link.getMethod()  : "") +
+                                 " resource with name: '" + link.getName() + "'";
+                    throw new BallerinaConnectorException(msg);
+                }
+            }
+        }
+    }
+
+    private static HttpResource getLinkedResource(HttpResource.LinkedResource link, HttpResource linkedResource,
+                                                  HttpResource resource) {
+        if (resource.getResourceName() != null && resource.getResourceName().equalsIgnoreCase(link.getName())) {
+            if (link.getMethod() != null) {
+                if (resource.getMethods() == null) {
+                    if (linkedResource == null) {
```

Comment:
Why we have neglected the else part here?

Reply:
`resource.getMethods` will return null when the resource method is `default`. So when we find a `default` resource for a linked resource with method, we update the linked resource only when there is no other resources with matching method. Therefore, for `default` resources, we only update the linked resource when it is null (this will avoid overwriting by `default` resource)